src/db/read.rs

```diff
@@ -90,22 +90,21 @@ mod tests {
     use crate::dag;
     use crate::db::write::init_db;
     use crate::kv::memstore::MemStore;
-    use crate::kv::Store;
     use crate::util::nanoserde::any::Any;
     use str_macro::str;
 
     #[async_std::test]
     async fn basics() {
-        let kv = MemStore::new();
-        init_db(&kv, "main", "local_create_date").await.unwrap();
-        let kvw = kv.write().await.unwrap();
-        let dw = dag::Write::new(kvw);
+        let ds = dag::Store::new(Box::new(MemStore::new()));
+        init_db(ds.write().await.unwrap(), "main", "local_create_date")
```

Comment:
I had to change the signature of `init_db` to be compatible with objects `connection.rs` has available (pass in a dag::Write instead of kv::Store).